src/types.ts

```diff
@@ -221,13 +217,52 @@ export type ChatCompletionOptionsFunction = {
   parameters: ObjectSchema;
 };
 
-type JSONSchema = (
-  | ObjectSchema
-  | StringSchema
-  | NumberSchema
-  | BooleanSchema
-  | ArraySchema
-) & { description?: string };
+interface SystemCompletionMessage {
+  content: string;
+  role: "system";
+}
+
+interface UserCompletionMessage {
+  content: string;
+  role: "user";
+}
+
+interface AssistantCompletionMessage {
+  content: string;
+  role: "assistant";
+}
+
+interface FunctionAwareAssistantCompletionMessage {
+  content: string | null;
+  role: "assistant";
+  function_call?: {
+    "name": string;
+    "arguments": string;
+  };
+}
+
+interface FunctionCompletionMessage {
+  content: string;
+  role: "function";
+  name: string;
+}
+
+export type ChatCompletionMessage =
+  | SystemCompletionMessage
+  | UserCompletionMessage
+  | FunctionAwareAssistantCompletionMessage
+  | FunctionCompletionMessage
+  | AssistantCompletionMessage;
```

Comment:
Here I'm summing both the "functional" version of the API and the "regular".  I thought about doing something more sophisticated (i.e. breaking out a type for "functional" models and altering the options here dynamically) but it started to head down a generic heavy path – which, in my opinion, complicated the typing more than it's worth.